pkg/command/root_test.go

```diff
@@ -908,3 +903,311 @@ func TestCompletionShortHelpInActiveHelp(t *testing.T) {
 
 	os.Unsetenv("TANZU_ACTIVE_HELP")
 }
+
+type fakePluginRemapAttributes struct {
+	name                 string
+	description          string
+	target               configtypes.Target
+	cmdGroup             plugin.CmdGroup
+	supportedContextType []configtypes.ContextType
+	invokedAs            []string
+	aliases              []string
+}
+
+func setupFakePluginInfo(p fakePluginRemapAttributes, pluginDir string) *cli.PluginInfo {
+	description := fmt.Sprintf("%s commands", p.name)
+	if p.description != "" {
+		description = p.description
+	}
+	cmdGroup := plugin.SystemCmdGroup
+	if p.cmdGroup != "" {
+		cmdGroup = p.cmdGroup
+	}
+
+	pi := &cli.PluginInfo{
+		Name:                 p.name,
+		Description:          description,
+		Group:                cmdGroup,
+		Version:              "v0.1.0",
+		Aliases:              []string{},
+		Hidden:               false,
+		InstallationPath:     filepath.Join(pluginDir, fmt.Sprintf("%s_%s", p.name, string(p.target))),
+		Target:               p.target,
+		InvokedAs:            []string{},
+		SupportedContextType: []configtypes.ContextType{},
+	}
+
+	if len(p.invokedAs) != 0 {
+		pi.InvokedAs = p.invokedAs
+	}
+	if len(p.supportedContextType) != 0 {
+		pi.SupportedContextType = p.supportedContextType
+	}
+	if len(p.aliases) != 0 {
+		pi.Aliases = p.aliases
+	}
+	return pi
+}
+
+// Tests behavior of command tree with plugins remapped (at a plugin level)
+func TestPluginLevelRemapping(t *testing.T) {
+	tests := []struct {
+		test              string
+		pluginVariants    []fakePluginRemapAttributes
+		args              []string
+		expected          []string
+		unexpected        []string
+		expectedFailure   bool
+		activeContextType configtypes.ContextType
+	}{
+		{
+			test: "one unmapped k8s plugin",
+			pluginVariants: []fakePluginRemapAttributes{
+				fakePluginRemapAttributes{
```

Comment:
The `fakePluginRemapAttributes` part of this line and the similar ones below give me a warning that it is a redundant declaration.  It seems that since the array type is specified, the element type need not be.